packages/flutter/test/cupertino/nav_bar_test.dart

```diff
@@ -883,6 +883,16 @@ void main() {
     );
     expect(SystemChrome.latestStyle, SystemUiOverlayStyle.dark);
   });
+
+  testWidgets('Back button shows an error when manually added outside a route', (WidgetTester tester) async {
```

Comment:
I think it would be helpful to explicitly refer to CupertinoNavigationBarBackButton in the test title to make it clear that we're talking about the use of a specific widget, rather than an incidental scenario where some kind of back button appears in the wrong place. E.g., "Exception thrown when CupertinoNavigationBarBackButton is declared outside of a Route."

Reply:
Also, given the cause of this issue, should there also be a test where the back button widget is used in a route, but it's a route that can't be popped?

Reply:
ya, good point. Added